### ISSUE TITLE
DRM-related installation issue

### ISSUE THREAD

**Issue by georgebisbas** [Syllo/nvtop#343]
Dear all, Hi!

First of all, thank you for this great tool. I/we have been using it for years, and we love it!
We are using nvtop in our CI/CD and it started to break the last 2 days.
We start with a brief story of how the bug started appearing in our CI and finish with a reproducible dockerfile
. I hope it helps and can reproduce the error on your side!

nvtop installs successfully up to commit:
(https://github.com/Syllo/nvtop/commit/9b7819ed8c9baea5fe4ff3d8b76c50477249fd6f)

The installation bug started to appear after merging this commit:
(https://github.com/Syllo/nvtop/commit/eddc07da7ea4fcfc91660db9f59750829d94a27c)

The complete error trace was:
```
7.702 /app/nvtop/src/extract_gpuinfo_intel_xe.c:31:10: fatal error: drm/xe_drm.h: No such file or directory
7.702    31 | #include <drm/xe_drm.h>
7.702       |          ^~~~~~~~~~~~~~
7.702 compilation terminated.
7.702 make[2]: *** [src/CMakeFiles/nvtop.dir/build.make:370: src/CMakeFiles/nvtop.dir/extract_gpuinfo_intel_xe.c.o] Error 1
7.702 make[1]: *** [CMakeFiles/Makefile2:143: src/CMakeFiles/nvtop.dir/all] Error 2
7.703 make: *** [Makefile:136: all] Error 2
```

The same error keeps appearing until this commit:
(https://github.com/Syllo/nvtop/commit/41023f2ace1d7e541cb1e9ba6170867c7fffdf9f)

obviously changing after merging:
(https://github.com/Syllo/nvtop/commit/56b78f1e8edd152db86429cb3d7d4e8aa8839f0c)

```c
extract_gpuinfo_intel_i915.c:110:57: error: 'DRM_I915_QUERY_MEMORY_REGIONS'
```
The complete trace is:

```bash
7.247 /app/nvtop/src/extract_gpuinfo_intel_i915.c: In function 'gpuinfo_intel_i915_refresh_dynamic_info':
7.247 /app/nvtop/src/extract_gpuinfo_intel_i915.c:110:57: error: 'DRM_I915_QUERY_MEMORY_REGIONS' undeclared (first use in this function)
7.247   110 |         intel_i915_query_alloc_fetch(gpu_info->card_fd, DRM_I915_QUERY_MEMORY_REGIONS, &length);
7.247       |                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
7.247 /app/nvtop/src/extract_gpuinfo_intel_i915.c:110:57: note: each undeclared identifier is reported only once for each function it appears in
7.247 /app/nvtop/src/extract_gpuinfo_intel_i915.c:112:39: error: invalid use of undefined type 'struct drm_i915_query_memory_regions'
7.247   112 |       for (unsigned i = 0; i < regions->num_regions; i++) {
7.247       |                                       ^~
7.247 /app/nvtop/src/extract_gpuinfo_intel_i915.c:113:16: error: variable 'mr' has initializer but incomplete type
7.247   113 |         struct drm_i915_memory_region_info mr = regions->regions[i];
7.247       |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
7.247 /app/nvtop/src/extract_gpuinfo_intel_i915.c:113:56: error: invalid use of undefined type 'struct drm_i915_query_memory_regions'
7.247   113 |         struct drm_i915_memory_region_info mr = regions->regions[i];
7.247       |                                                        ^~
7.247 /app/nvtop/src/extract_gpuinfo_intel_i915.c:113:44: error: storage size of 'mr' isn't known
7.247   113 |         struct drm_i915_memory_region_info mr = regions->regions[i];
7.247       |                                            ^~
7.250 /app/nvtop/src/extract_gpuinfo_intel_i915.c:115:39: error: 'I915_MEMORY_CLASS_DEVICE' undeclared (first use in this function)
7.250   115 |         if (mr.region.memory_class == I915_MEMORY_CLASS_DEVICE || regions->num_regions == 1) {
7.250       |                                       ^~~~~~~~~~~~~~~~~~~~~~~~
7.250 /app/nvtop/src/extract_gpuinfo_intel_i915.c:115:74: error: invalid use of undefined type 'struct drm_i915_query_memory_regions'
7.250   115 |         if (mr.region.memory_class == I915_MEMORY_CLASS_DEVICE || regions->num_regions == 1) {
7.250       |                                                                          ^~
7.254 make[2]: *** [src/CMakeFiles/nvtop.dir/build.make:356: src/CMakeFiles/nvtop.dir/extract_gpuinfo_intel_i915.c.o] Error 1
7.254 make[1]: *** [CMakeFiles/Makefile2:143: src/CMakeFiles/nvtop.dir/all] Error 2
```

To reproduce the error you may use the following Dockerfile.nvtop:

```docker
##############################################################
# Minimal failing example for syllo/nvtop installation issue 
##############################################################

FROM ubuntu:22.04 AS base

ENV DEBIAN_FRONTEND=noninteractive

# Install essentials
RUN apt-get update && \
    apt-get install -y dh-autoreconf && \
    apt-get install -y wget git flex hwloc \
        autoconf libtool build-essential

# Update if outdated
RUN apt-get update

# Install Nvtop
RUN apt-get install -y git cmake libncurses5-dev libncursesw5-dev libdrm-dev libsystemd-dev cmake && \
    git clone https://github.com/Syllo/nvtop.git /app/nvtop && \
    cd /app/nvtop && \
    git checkout c9bfedcc3a20a7d8309024ccb7a3f01bf7de17db && \
    cd ../../ && \
    mkdir -p /app/nvtop/build && cd /app/nvtop/build && \
    cmake .. -DNVIDIA_SUPPORT=ON -DAMDGPU_SUPPORT=ON -DINTEL_SUPPORT=ON && \
    make && \
    make install

# Safety cleanup
RUN apt-get clean && apt-get autoclean && apt-get autoremove -y && \
    rm -rf /var/lib/apt/lists/*
```

The error should be reproduced by:
```
docker build . --file docker/Dockerfile.nvtop
```
You may change the commit hash to cross-check with other commits.

**Comment by Syllo** [comment]
Hello,

Thanks for the bug report, it was easy for me to reproduce!

Ubuntu 22.04 ships with a version of `libdrm-dev` that is a little too old (2.4.110 and requires >= 2.4.114) to have the new i915 memory query bits.
I did a patch that adds the struct declarations and macros that are missing. It seems to fix the issue (tested on Ubuntu 22.04 and 20.04).

**Comment by georgebisbas** [comment]
Thanks for the quick fix, it seems that our workflows are passing again now!
